Submitted By:            Randy McMurchy <randy_at_linuxfromscratch_dot_org>
Date:                    2005-05-26
Initial Package Version: 1.2.1
Upstream Status:         Reported
Origin:                  Randy McMurchy - information found at:
                         http://www.sleepycat.com/docs/ref/upgrade.4.3/stat.html
Description:             Adds a check for BDB >= version 4.3 and adds the required
                         API change if necessary

diff -Nau libgda-1.2.1-orig/providers/bdb/gda-bdb-recordset.c libgda-1.2.1/providers/bdb/gda-bdb-recordset.c
--- libgda-1.2.1-orig/providers/bdb/gda-bdb-recordset.c	2003-12-17 11:27:02.000000000 +0000
+++ libgda-1.2.1/providers/bdb/gda-bdb-recordset.c	2005-05-26 20:08:08.000000000 +0000
@@ -227,6 +227,9 @@
 
 	/* get the number of records in the database */
 	ret = dbp->stat (dbp,
+#if BDB_VERSION >= 40300
+			 NULL,
+#endif
 			 &statp,
 #if BDB_VERSION < 40000
 			 NULL,
